### PR TITLE
Develop #57 Employment trends

### PR DESCRIPTION
**Pull request by rpitingolo**

### CODE REVIEW

**Comment by rpitingolo** [comment]
@ptatian I should have put my comment over here. Copy and pasted over from the issue.

Can you please review Employment Trends.sas which are the BLS data indicators. There are some weird things going on.

On sectorjobs_byyear there are no reported jobs in state and local government in any years. The big file even shows establishment counts but no employment counts for state and local government. I'm not sure what's going on there.

On industryjobs_byyear, the trend seems reasonable (have there really been 750k new jobs added since 1990?) but the actual numbers by naics code seem off. For example construction jumps from zero to 174k then back down to zero. I also can't get the numbers to sum up to the total, even when taking out the 101 and 102 naics codes. Any ideas?

**Comment by ptatian** [comment]
Thanks, @rpitingolo, for flagging the issues. I will take a look.

**Comment by ptatian** [comment]
@rpitingolo For everything except the first item (jobs for 15 MSAs), you should be aggregating employment for the **RHF region** from counties, not using the MSA level. I think this is why you are having the problem. 

When you do this, please check the coverage of the individual counties. It looks like there are gaps for some counties in earlier years so if we can't reliably go back to 1990 for all counties we can start later.

**Comment by rpitingolo** [comment]
@ptatian thanks, I'll try to make that change and let you know about the earlier years.

**Comment by rpitingolo** [comment]
@ptatian it looks like we are OK to go back to 1990. There is a lot of missing data for Manassas Park but it's in newer years too and I'm not sure there is anything we can do about that.

**Comment by ptatian** [comment]
Thanks, @rpitingolo. That sounds fine to proceed in that manner.

**Comment by ptatian** [comment]
@rpitingolo Are you ready for me to look at the changes you made?

**Comment by rpitingolo** [comment]
@ptatian yes, ready for you to look at. Please note that I had to recode the Los Angeles MSA because it looks like the code changed in 2013.

File: Prog/Employment Trends.sas [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+proc summary data = msajobs;
+	class Area;
+	var Annual_Average_Employment_: ;
+	output out = msajobs_t sum=;
+run;
+
+data msajobs_byyear;
+	set msajobs_t;
+	drop _type_ _freq_;
+	if area = " " then area = "Total";
```
Comment (ptatian) [inline Prog/Employment Trends.sas]: I did say to always include a total row but, for this table, it isn't needed. You can just delete the total row here.

File: Prog/Employment Trends.sas [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+** Define libraries **;
+%DCData_lib( BLS )
+%DCData_lib( RegHsg )
+
+/* Update parameters for data updates */
+%let start_yr = 1990;
+%let end_yr = 2017;
+
+/* List of MSAs for comparison */
+%let msalist = 	"C1206", 	/* Atlanta-Sandy Springs-Roswell */
```
Comment (ptatian) [inline Prog/Employment Trends.sas]: I like the use of comments here to label the values. :100:

File: Prog/Employment Trends.sas [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+proc export data = msajobs_byyear
+	outfile = "&_dcdata_default_path.\reghsg\prog\msajobs_byyear.csv"
+	dbms=csv
+	replace;
+run;
+
+
+
+/* Jobs by private, federal gov't, local/state gov't, 1990 - 2017 */
+data sectorjobs;
```
Comment (ptatian) [inline Prog/Employment Trends.sas]: Please combine state and local govt jobs into a single category.

**Comment by rpitingolo** [comment]
@ptatian Employment Trends.sas have been addressed. I think it is ready for finalizing.

iPums Employment Trends.sas is ready for review. Please check out the numbers on wage_byyear.csv in particular. I know 2010 was during a recessiong but the change in FT workers is huge. The code is macro'd so I don't think there is a coding error and it looks like weights are being applied in the right places. Thanks.

**Comment by ptatian** [comment]
Thanks, @rpitingolo. This looks good to finalize. I wasn't sure which table you were referring to (you said wage_byyear.csv) but I didn't see anything alarming in the overall job trends.

**Comment by rpitingolo** [comment]
@ptatian thanks. If the numbers look right to you I'll go ahead and finalize.

**Pull request merged**